### DISCUSSION TITLE
Arithmetic operator precedence changed between version 0.12.1 and 0.13.0?

### DISCUSSION THREAD

**Discussion by yump** [bpftrace/bpftrace#2095] [General]
With bpftrace 0.12.1 packaged in Fedora 34, and kernel version 5.15.8, multiplication is applied before addition, in the usual way:

```
$ bpftrace --version
bpftrace v0.12.1
$ sudo bpftrace -e 'BEGIN {printf("%d\n", (10 + 1 * 5)); }'
Attaching 1 probe...
15
^C
```

But using packaged bpftrace 0.13.0 on Fedora 35, with the same kernel version, the expression is evaluated left-to-right:

```
$ bpftrace --version
bpftrace v0.13.0
$ sudo bpftrace -e 'BEGIN {printf("%d\n", (10 + 1 * 5)); }'
Attaching 1 probe...
55
^C
```

Because it seems so unlikely that this could change without anyone else noticing, I'm a bpf novice, and I haven't set up an environment to build bpftrace from source and bisect it, it didn't feel right to report this as a bug.

**Comment by fbs** [comment]
no, sadly you're right. I think it went unnoticed as its not a common thing to do (IMO). I'll get a fix out

**Reply by fbs** [reply]
#2096 should fix it

**Comment by viktormalik** [comment]
bpftrace 0.13.1 fixing this issue has just been released to Fedora 35.
